File: UITableViewThreeEventsDemo/Demo-objc/Demo-objc/DelegateCell.m
//
//  DelegateCell.m
//  Demo-20160419
//
//  Created by crw on 4/19/16.
//  Copyright © 2016 crw. All rights reserved.
//

#import "DelegateCell.h"

@implementation DelegateCell

- (void)setModel:(Model *)model{
    _model = model;
    
    self.titleLabel.text = model.title;
}

- (void)setDelegate:(id<CellDelegate>)delegate{
    _delegate = delegate;
}

- (IBAction)delClick:(UIButton *)sender {
    if ([self.delegate respondsToSelector:@selector(delBtnClick:)]) {
        [self.delegate delBtnClick:self];
    }
}
@end

File: UITableViewThreeEventsDemo/Demo-swift/Demo-swift/DelegateCell.swift
//
//  DelegateCell.swift
//  Demo-20160418
//
//  Created by crw on 4/20/16.
//  Copyright © 2016 crw. All rights reserved.
//

import UIKit

protocol CellDelegate {
    func delBtnClick(cell : DelegateCell )
}

class DelegateCell: UITableViewCell {
    
    @IBOutlet weak var titleLabel: UILabel!
    
    var delegate : CellDelegate?
    
    var model : Model? {
        didSet{
            if let model = model {
                titleLabel.text = model.title
            }
        }
    }
    
    @IBAction func delClick(sender: UIButton) {
        delegate?.delBtnClick(self)
    }
    
}
